### PR TITLE
Make ServiceCollection TryAdd() extension methods chainable

### PR DESCRIPTION
**Pull request by tgharold**
The "Add...()" and "Replace()" extension methods on `IServiceCollection` are already chainable, but the "TryAdd...()" methods are not.  This changes the return type to `public IServiceCollection` to line up with the other methods and adds the return statements.

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/Extensions/ServiceCollectionDescriptorExtensions.cs [hunk 1]
```diff
@@ -71,7 +71,8 @@ public static class ServiceCollectionDescriptorExtensions
         /// </summary>
         /// <param name="collection">The <see cref="IServiceCollection"/>.</param>
         /// <param name="descriptor">The <see cref="ServiceDescriptor"/> to add.</param>
-        public static void TryAdd(
+        /// <returns>A reference to the current instance of <see cref="IServiceCollection"/>.</returns>
+        public static IServiceCollection TryAdd(
```
Comment (gfoidl) [inline src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/Extensions/ServiceCollectionDescriptorExtensions.cs:75]: Just to point it out: this is a breaking change, as the return type of the method is changed.

Comment (tgharold) [reply src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/Extensions/ServiceCollectionDescriptorExtensions.cs:75]: Existing C# code would not be affected as no existing C# code is using these methods in a chaining fashion.  But I'm not sure about any interaction with non-C# code.

I did not change the `/ref` folder files yet which is why some checks failed.

I suspect that I'll need to close this and submit an [API review issue](https://github.com/dotnet/runtime/blob/master/docs/project/api-review-process.md) in order to get the change considered.

**Comment by davidfowl** [comment]
This is a breaking change, we’re not going to take it.

**Comment by stephentoub** [comment]
Thanks for the PR.  Closing as this is a binary breaking change.

**Pull request closed without merging**